src/Accounts/Authentication/Utilities/AzConfigReader.cs

```diff
@@ -31,22 +31,50 @@ private static IAzureSession Session
             }
         }
 
+        private static IConfigManager instance = null;
+
+        public static IConfigManager Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    if (!Session.TryGetComponent<IConfigManager>(nameof(IConfigManager), out instance))
+                    {
+                        instance = null;
+                    }
+                }
+                return instance;
+            }
+        }
+
         public static T GetAzConfig<T>(string key, T defaultValue = default(T))
         {
-            return Session.TryGetComponent<IConfigManager>(nameof(IConfigManager), out IConfigManager configManager) ? configManager.GetConfigValue<T>(key) : defaultValue;
+            return Instance != null ? Instance.GetConfigValue<T>(key) : defaultValue;
+        }
+
+        public static void DisableWam()
```

Comment:
I feel like a config "Reader" should not be able to write config values, but as long as the token cache issue is fixed this PR will be reverted (correct me if I'm wrong) so it's probably OK for now.